<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
       //声明两个Promise对象
       const p1 = new Promise((resolve,reject)=>{
        setTimeout(()=>{
          resolve('商品数据 - 1');
          
        },1000)
       });

      //  const p2 = new Promise((resolve,reject)=>{
      //   setTimeout((error)=>{
      //     reject(error)
      //   },1000)
      //  });
       const p2 = new Promise((resolve,reject)=>{
        setTimeout(()=>{
          // reject(error)
          resolve('商品数据 - 2')
        },1000)
       });

       //调用 allsettled 方法 即便两个都失败 返回的也是成功
       //此时如果调用 promise.all方法 其中一个primise失败后立即失败
       const result = Promise.allSettled([p1,p2]);
      //  const result = Promise.all([p1,p2]);
       console.log(result);

       
       
  </script>
</body>
</html>